addons/auth_oauth/models/res_users.py

```diff
@@ -65,7 +93,7 @@ def _auth_oauth_signin(self, provider, validation, params):
 
             This method can be overridden to add alternative signin methods.
         """
-        oauth_uid = validation['user_id']
+        oauth_uid = validation['sub']
```

Comment:
@xmo-odoo Using standards is great, thanks for this change.

However it looks like this line is breaking compatibility, in particular with OCA `auth_oidc`, where users configure a mapping to the `user_id` field. Since `_auth_oauth_validate` also sets the `user_id` field, this change to look for `sub` should be done in master only, IMO.

Reply:
Fair go, it looks like we didn't think of that when I was reminded to re-set `user_id` in case an extension *used* the oauth id. I created #91941

Reply:
Thanks a lot for taking care @xmo-odoo !